app/controllers/assignments_controller.rb

```diff
@@ -1,5 +1,6 @@
 class AssignmentsController < ApplicationController
   before_action :admin?
+  before_action :logged_in_user, only: [:new, :create]
```

Comment:
make this filter to apply to all the actions in assignments_controller.
Only logged in users can access any action in assignments